mgmt-lambda/handlers/updateHandler.ts

```diff
@@ -35,7 +38,25 @@ export async function handleUpdate(
     throw new ApiException(ErrorCode.LambdaFunctionNotFound)
   }
 
+  const listVersionsBeforeUpdate = await listLambdaFunctionVersions(lambdaClient, settings.LambdaFunctionName)
   const functionVersionArn = await updateLambdaFunctionCode(lambdaClient, settings.LambdaFunctionName)
+  const listVersionsAfterUpdate = await listLambdaFunctionVersions(lambdaClient, settings.LambdaFunctionName)
+
+  const newVersions = Array.from(listVersionsAfterUpdate.values()).filter(
+    (conf) => !listVersionsBeforeUpdate.has(conf.Version)
+  )
+  if (newVersions.length !== 1) {
```

Comment:
Could you explain why we need this check? At this point, we've called `updateLambdaFunctionCode` with no exceptions, so the lambda function is successfully updated. This check, if fails, will throw exception even though we've successfully updated the lambda function, right? @sshelomentsev

Reply:
In that case, what is our solution? For instance, do you think waiting for 5 seconds and then re-trying helps? This is an HTTP handler, so we should not wait for a long time (max 30 secs in total), I am aware, but the problem is that, if management lambda returns an error during update, this will be shown to the user in the dashboard. What should the customer do in that case? If they need to try again, we could just try again in the code, is my point.

Reply:
We can have the "try again logic" in the management lambda or in the dashboard backend. Both are better than current implementation IMO

Reply:
Never mind. Update problems should not affect the customer's user experience in the dashboard.